### PR TITLE
IRAP-462 -> main

### PR DESCRIPTION
**Pull request by SamDudley**

### CODE REVIEW

File: core/views/irap_request.py [hunk 1]
```diff
@@ -44,6 +55,24 @@ def get_context_data(self, **kwargs):
...
+    def get_step_status(self, irap_id) -> IrapStepStatus:
+        # IrapRequestStepView is a step in a task.
+        # A step can be not relevant, completed or not started.
+        # The step 'relevancy' is calculated from the irap_request values already entered
+        # If the related form return is_valid = True, the step is completed
+        # TODO error for not valid id
+        irap_dict = model_to_dict(self.model.objects.get(pk=irap_id))
+        form = self.form_class(irap_dict)
+        if self.is_relevant is False:
+            return IrapStepStatus.Not_Relevant
```
[the remark this answers is not in the document]
Comment (Luisella21) [reply core/views/irap_request.py]: Done.

File: core/views/task_list.py [hunk 2]
```diff
@@ -1,7 +1,113 @@
...
+    # or a list of tasks and steps
+    # This covers 'steps' that consist of more of one form,
+    name: str
+    page_name = "undefined"
+    steps = []
+    _step_count = 0
+    _complete_count = 0
+    _status_list = []
+    _calculated = False
+    _task_status: IrapStepStatus = IrapStepStatus.Not_Started
```
[the remark this answers is not in the document]
Comment (Luisella21) [reply core/views/task_list.py]: Class changed completely.

File: core/views/task_list.py [hunk 3]
```diff
@@ -2,6 +2,102 @@
...
+
+irap_tasks = {
+    "ProjectDetails": ProjectDetailTask,
+    "ContactDetail": ContactDetailTask,
+    "ProjectTeam": ProjectTeamTask,
+    "Timeline": TimelineTask,
+    "UseOfData": UseOfDataTask,
+}
+
+
```
Comment (SamDudley) [inline core/views/task_list.py:83]:
```python
from core.templatetags.gds import gds_tag

complete_tag = gds_tag("complete")
incomplete_tag = gds_tag("incomplete", "gray")
...

status_to_tag = {
    IrapStepStatus.COMPLETE: complete_tag,
    ...
}

tag = status_to_tag[status]
```

File: core/views/irap_request.py [hunk 4]
```diff
@@ -12,6 +14,14 @@
 from core.services.irap_request import irap_request_submitted
 
 
+class IrapStepStatus(Enum):
+    COMPLETED = "complete"
+    NOT_STARTED = "not started"
+    STARTED = "started"
```
Comment (SamDudley) [inline core/views/irap_request.py:20]: move to task status

File: core/views/irap_request.py [hunk 5]
```diff
@@ -12,6 +14,14 @@
...
+    NOT_RELEVANT = "not relevant"
+    CANNOT_START = "cannot start"
```
Comment (SamDudley) [inline core/views/irap_request.py:22]: move to task status

File: core/views/task_list.py [hunk 6]
```diff
@@ -2,6 +2,102 @@
...
+    IrapStepStatus,
+    ProjectTeam,
+    RequestType,
+    Timelines,
+    UsingData,
+)
+
+
+class IrapGenericTask:
+    steps: [IrapStepStatus]
```
Comment (SamDudley) [inline core/views/task_list.py:17]:
```python
steps: list[IrapRequestStepView]
```

File: core/views/task_list.py [hunk 7]
```diff
@@ -2,6 +2,102 @@
...
+
+    def calculate_status(self):
```
Comment (SamDudley) [inline core/views/task_list.py:19]:
```python
def calculate_status(self) -> IrapTaskStatus:
```

File: core/views/task_list.py [hunk 8]
```diff
@@ -2,6 +2,102 @@
...
+irap_tasks = {
+    "ProjectDetails": ProjectDetailTask,
+    "ContactDetail": ContactDetailTask,
+    "ProjectTeam": ProjectTeamTask,
+    "Timeline": TimelineTask,
+    "UseOfData": UseOfDataTask,
+}
+
+
+def create_task_labels(irap_obj) -> dict:
```
Comment (SamDudley) [inline core/views/task_list.py:84]:
```python
def create_task_labels(irap_obj) -> dict[str, Any]:
```

File: core/templates/core/irap_request/task_list.html [hunk 9]
```diff
@@ -59,25 +59,23 @@ <h2>Collaborators</h2>
...
-    <li>
-        <h2>1. Your request</h2>
+    <h2>1. Details</h2>
+
         <ul>
-            <li><a href="{% url 'core:request:request-type' object.pk %}">Project details</a>{{ complete_tag }}</li>
-            <li><a href="{% url 'core:request:contact-details' object.pk %}">Contact details</a>{{ complete_tag }}</li>
-            <li><a href="{% url 'core:request:project-team' object.pk %}">Project team</a>{{ complete_tag }}</li>
-            <li><a href="{% url 'core:request:timelines' object.pk %}">Timelines and dependencies</a>{{ complete_tag }}</li>
+            <li><a href="{% url 'core:request:request-type' object.pk %}">Project details</a>{% gds_tag tasks.task_labels.ProjectDetails.value %}</li>
```
Comment (SamDudley) [inline core/templates/core/irap_request/task_list.html:71]:
```
{{ tasks.task_labels.ProjectDetails }}
```

File: core/views/irap_request.py (hunk 4 above)
Comment (Luisella21) [reply core/views/irap_request.py:20]: Done.

File: core/views/task_list.py (hunk 3 above)
Comment (Luisella21) [reply core/views/task_list.py:114]: Done. It works!

File: core/views/task_list.py [hunk 10]
```diff
@@ -1,7 +1,113 @@
...
+    _calculated = False
+    _task_status: IrapStepStatus = IrapStepStatus.Not_Started
+
+    def __init__(self, irap_pk=0):
+        self.irap_pk = irap_pk
+
+    def calculate_status(self):
+        if self._calculated:
+            # Avoid recalculating it the list
+            return
```
[the remark this answers is not in the document]
Comment (Luisella21) [reply core/views/task_list.py]: Changed the logic.

**Pull request merged**